src/services/sanity.js

```diff
@@ -1374,6 +1374,77 @@ export async function fetchGenreLessons(brand, name, contentType, {
   return fetchSanity(query, true);
 }
 
+export async function fetchTopLevelParentId(railcontentId) {
+    const query = `*[railcontent_id == ${railcontentId}]{
+      railcontent_id,
+      'parents': *[^._id in child[]._ref && !(_id in path('drafts.**'))]{
+        railcontent_id,
+          'parents': *[^._id in child[]._ref && !(_id in path('drafts.**'))]{
+            railcontent_id,
+            'parents': *[^._id in child[]._ref && !(_id in path('drafts.**'))]{
+              railcontent_id,
+               'parents': *[^._id in child[]._ref && !(_id in path('drafts.**'))]{
+                  railcontent_id,               
+            } 
+          }
+        }
+      }
+    }`;
+    let response = await fetchSanity(query, false, {processNeedAccess: false});
+    if (!response) return null;
+    let currentLevel = response;
+    for (let i = 0; i < 4; i++) {
+        if (currentLevel['parents'].length > 0) {
+            currentLevel = currentLevel['parents'][0];
+        } else {
+            return currentLevel['railcontent_id'];
+        }
+    }
+    return null;
+}
+
+export async function fetchHierarchy(railcontentId) {
+    let topLevelId = await fetchTopLevelParentId(railcontentId);
+    const query = `*[railcontent_id == ${topLevelId}]{
```

Comment:
Same thought on if we can get this data directly from the Sanity data, and if not, if it should be configurable for how many levels we go to.

Reply:
This wasn't trivial so i did not do it.  Also the intention here is to have the full hierarchy pulled down in one shot so we can make use of the caching for lots of different ids.